mindmeld/components/entity_resolver.py

```diff
@@ -42,60 +50,277 @@
     load_index,
     resolve_es_config_for_version,
 )
+from ._util import _is_module_available, _get_module_or_attr
+from .. import path
+from ..core import Entity
+from ..exceptions import EntityResolverConnectionError, EntityResolverError
 
 logger = logging.getLogger(__name__)
 
 
-class EntityResolver:
-    """An entity resolver is used to resolve entities in a given query to their canonical values
-    (usually linked to specific entries in a knowledge base).
-    """
+def _correct_deprecated_er_config(er_config):
+    # for deprecated usage of er_config; code snippet to maintain backwards compatability
+    #   if `er_config` is supplied in deprecated format, it is modified to correct format and
+    #   returned, else it is not modified
 
-    # prefix for Elasticsearch indices used to store synonyms for entity resolution
-    ES_SYNONYM_INDEX_PREFIX = "synonym"
-    """The prefix of the ES index."""
+    if not er_config.get("model_settings", {}).get("resolver_type", None):
+        model_type = er_config.get("model_type")
+        if model_type == "resolver":
+            raise Exception("Could not find `resolver_type` in `model_settings` of entity resolver")
```

Comment:
Not sure what this check if doing? Its ok for `resolver_type ` to be in the new API but not in the old?

Reply:
Its catching the exception when the the configs are following new format, but do not have `resolver_type` in its model_settings.

Reply:
If configs are following old format, then the `model_type` won't be `resolver`, in which case the `else` condition is hit and a deprecation warning is raised. In the old format, one need not provide any `model_settings` and just `model_type` suffices.

Reply:
Got it.